app/models/user.rb

```diff
@@ -36,13 +37,28 @@ def forget
     update_attributes :remember_digest, nil
   end
 
-  def authenticated? remember_token
-    return false unless remember_digest
-    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  def authenticated? attribute, token
+    digest = send("#{attribute}_digest")
+    return false unless digest
+    BCrypt::Password.new(digest).is_password? token
```

Comment:
đã ai hỏi em cái is_password? chạy thế nào chưa?

Reply:
Em chưa được hỏi ợ.

Reply:
chưa được hỏi thì rep đi em

Reply:
+ Hàm is_password là alias của toán tử so sánh ==. 
+ BCrypt::Password kế thừa từ String.
+ Khi gọi phương thức is_password, bản chất là thực hiện phép so sánh chuỗi giữa token nhập vào và 1 chuỗi đã được hash.
+ Hàm is_password sẽ gọi tiếp phương thức == của String, thực hiện so sánh chuỗi đã được hash với một chuỗi hash mới được sinh ra từ token nhập vào và @salt.